react/new/src/contexts/ThemeContext.tsx

```diff
@@ -0,0 +1,156 @@
+import {
+  createContext,
+  useContext,
+  useCallback,
+  useState,
+  ReactNode,
+} from 'react'
+import {
+  MantineProvider,
+  createTheme,
+  type MantineColorsTuple,
+} from '@mantine/core'
+import { NHL_TEAM_COLORS } from '../constants/nhlColors'
+import { ThemeConfig, ThemeMode, NHLTeam } from '../types/theme'
+
+interface ThemeContextType {
+  theme: ThemeConfig
+  setThemeMode: (mode: ThemeMode) => void
+  setTeamTheme: (team: NHLTeam | undefined) => void
+}
+
+const ThemeContext = createContext<ThemeContextType | undefined>(undefined)
+
+export const useTheme = () => {
+  const context = useContext(ThemeContext)
+  if (!context) {
+    throw new Error('useTheme must be used within a ThemeProvider')
```

Comment:
When ts throws errors what happens. Do they bubble back to the top for us to handle? Do they automatically throw and alert?

Reply:
The error should show up in the dev console unless it's suppressed in some way, which it shouldn't be.